### PR TITLE
Push button polling cleanup

### PR DESCRIPTION
**Pull request by adrianstevens**

### CODE REVIEW

File: Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs [hunk 1]
```diff
@@ -67,32 +60,33 @@ public bool State
...
 
         /// <summary>
-        /// Raised when the button circuit is re-opened after it has been closed (at the end of a press)
+        /// Raised when the button is released after a press
         /// </summary>
         public event EventHandler Clicked = delegate { };
 
         /// <summary>
-        /// Raised when the button is pressed for LongClickedThreshold or longer
+        /// Raised when the button is pressed for LongClickedThreshold or longer and then releases
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs:78]: how about "Raised when the button is released after being pressed for LongClickedThreshold or longer"?

File: Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs [hunk 2]
```diff
@@ -134,24 +128,23 @@ public PushButton(IDigitalInputPort inputPort)
...
             else
-            {   //if we don't have full interrupts, fall back to polling
-                //Re set the resistor mode if Interrupt mode is none - ToDo - known issue
-                if ((DigitalIn.InterruptMode == InterruptMode.None && resistorMode == ResistorMode.InternalPullUp) ||
-                    (DigitalIn.InterruptMode == InterruptMode.None && resistorMode == ResistorMode.InternalPullDown))
-                { DigitalIn.Resistor = resistorMode; }
-
+            {
+                //ToDo remove resistor mode hack for RC2
+                Console.WriteLine($"Resistor mode hack {resistorMode}");
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs:138]: Will we want this console output when we release?

File: Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs [hunk 3]
```diff
@@ -164,15 +157,39 @@ public PushButton(IDigitalInputPort inputPort)
...
+        {
+            return (resistorMode == ResistorMode.InternalPullUp ||
+                    resistorMode == ResistorMode.ExternalPullUp) ? !state : state;
         }
 
         void UpdateEvents(bool state)
         {
-            state = (resistorMode == ResistorMode.InternalPullUp ||
-              resistorMode == ResistorMode.ExternalPullUp) ? !state : state;
+            Console.WriteLine($"State: {state}, mode: {resistorMode}");
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs:192]: Remove output when we release?

File: Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs (hunk 2 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs]: haha nope ... deleted :)

File: Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs (hunk 3 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Core/Sensors/Buttons/PushButton.cs]: yup - done!

**Review by jorgedevs — APPROVED** [review]

**Pull request merged**